calico_node/filesystem/etc/service/felix/run

```diff
@@ -1,5 +1,10 @@
 #!/bin/sh
 exec 2>&1
+# Felix doesn't understand HOSTNAME, but the container exports it as a common
+# interface. This ensures Felix gets the right hostname.
+if [ ! -z $HOSTNAME ]; then
+    export FELIX_FELIXHOSTNAME=$HOSTNAME
```

Comment:
I considered that, but since Felix evolves independently it didn't seem wise to potentially set the environment variable to an empty string rather then leaving it unset when we're not trying to control it.